### PR TITLE
Ayowel/feature/more options

### PR DESCRIPTION
**Pull request by Ayowel**
To have the same result as before (new filename extension) 
`new filename -e extension`

* It is now possible to generate several files at once
* It is now possible to generate sets of files with the same name but differents extensions (.c/.h, ...)
* It is now possible to pass arguments to the builder (potentially allows to generate class/interface depending on context instead of extension only)
* It is now possible to chose which builder will be used (with some tolerance on the extension, actually allows the full builder name or main name if builder in .sh or .py)
* it is now possible to consult the help in new

Moreover, extending the options should not prove too hard if needed, and the error codes are now formalised at the start of the file

### CODE REVIEW

**Comment by Ricain** [comment]
Hi, thank for your contribution.
I will merge but first I need to review, so I come back to you in a few days.

**Comment by Ayowel** [comment]
Sure.

Didn't type it before, but there is two ways to type short & long parameters:

```
new file -ec
new file -e c
new file --extension=c
new file --extension c
```

File: new [hunk 1]
```diff
@@ -8,30 +8,241 @@
...
+			file_not_generated=$[file_not_generated+1]
+			continue
+		fi
+
+		builder_found=0
+		for build_extension in "${BUILDER_EXTENSIONS[@]}"; do
+			if [ -f "$loc/$extension/$default$build_extension" ]; then
+				output="$("$loc/$extension/$default$build_extension" "$filename" "$loc/$extension")"
+				if [ ! $? ]; then
+					file_builder_error=$[file_builder_error+1]
```
Comment (Ricain) [inline new]: There is a small bug on line 227: it should be `file_builder_error` with a `s` like defined on line 191 and 245.

I will correct it in a next commit 😄

**Comment by Ricain** [comment]
I will merge, thank for your contribution 👍

**Pull request merged**